recipes/flash-attn/meta.yaml

```diff
@@ -0,0 +1,78 @@
+{% set name = "flash-attn" %}
+{% set version = "2.5.8" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/flash_attn-{{ version }}.tar.gz
+    sha256: 2e5b2bcff6d5cff40d494af91ecd1eb3c5b4520a6ce7a0a8b1f9c1ed129fb402
+  # Overwrite with a simpler build script that doesn't try to revend pre-compiled binaries
+  - path: pyproject.toml
+  - path: setup.py
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vvv --no-deps --no-build-isolation
+  script_env:
+    # Temporarily reduce ARCHs and JOBS to debug build
+    # - MAX_JOBS=$CPU_COUNT
+    - MAX_JOBS=1
+    # - TORCH_CUDA_ARCH_LIST=8.0;8.6;8.9;9.0+PTX
+    - TORCH_CUDA_ARCH_LIST=8.6+PTX
```

Comment:
Set MAX_JOBS=1 again as in ef03f903d3195d459d95f63946b615336c33d96f. Looks like CI can run up to 6 hours now, though that's still not enough to finish compiling :sweat_smile: See e.g. https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=928208&view=logs&jobId=67448ffb-e003-5bfa-c062-cee3af60fcba&j=4f860608-e5f8-5c9c-4eb0-308a99ecb61e&t=02ef1a5c-d960-5c54-fcea-983775f057bb where the CUDA 11.8 build got as far as 30/49.

Reply:
Oo, it looks the compilation finally finished, at least a single Python version (Python 3.11). The CI check shows that the job was cancelled after 6 hours, but it's because it continued to try to build for another Python version (Python 3.10) for some reason (maybe because we're not using noarch).

Logs from https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=928722&view=logs&j=67448ffb-e003-5bfa-c062-cee3af60fcba&t=818ff20d-11b7-59db-6ce1-bb4df921454a&l=1226 showing successful build

<details>

```python-traceback
[49/49] /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/bin/nvcc  -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/csrc/flash_attn -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/csrc/flash_attn/src -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/csrc/cutlass/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.11/site-packages/torch/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.11/site-packages/torch/include/torch/csrc/api/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.11/site-packages/torch/include/TH -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.11/site-packages/torch/include/THC -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/include/python3.11 -c -c /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/csrc/flash_attn/src/flash_fwd_split_hdim96_fp16_sm80.cu -o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim96_fp16_sm80.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ --expt-relaxed-constexpr --expt-extended-lambda --use_fast_math -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1017"' -DTORCH_EXTENSION_NAME=flash_attn_2_cuda -D_GLIBCXX_USE_CXX11_ABI=1 -gencode=arch=compute_86,code=compute_86 -gencode=arch=compute_86,code=sm_86 -ccbin /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/bin/x86_64-conda-linux-gnu-cc
2024-05-07T07:10:57.5843417Z   nvcc warning : incompatible redefinition for option 'compiler-bindir', the last value of this option was used
2024-05-07T07:10:57.7918559Z   /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/bin/x86_64-conda-linux-gnu-c++ -shared -Wl,--allow-shlib-undefined -Wl,-rpath,/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -Wl,-rpath-link,/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -Wl,--allow-shlib-undefined -Wl,-rpath,/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -Wl,-rpath-link,/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -Wl,-rpath-link,/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib/stubs -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/lib/stubs -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/include -fdebug-prefix-map=/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work=/usr/local/src/conda/flash-attn-2.5.8 -fdebug-prefix-map=/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_=/usr/local/src/conda-prefix -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/include -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib/stubs -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/lib/stubs -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/include -I/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/include -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/targets/x86_64-linux/lib/stubs -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/targets/x86_64-linux/lib/stubs /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/flash_api.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim128_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim128_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim160_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim160_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim192_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim192_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim224_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim224_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim256_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim256_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim32_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim32_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim64_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim64_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim96_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_bwd_hdim96_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim128_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim128_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim160_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim160_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim192_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim192_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim224_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim224_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim256_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim256_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim32_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim32_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim64_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim64_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim96_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_hdim96_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim128_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim128_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim160_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim160_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim192_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim192_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim224_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim224_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim256_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim256_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim32_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim32_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim64_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim64_fp16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim96_bf16_sm80.o /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work/build/temp.linux-x86_64-cpython-311/csrc/flash_attn/src/flash_fwd_split_hdim96_fp16_sm80.o -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.11/site-packages/torch/lib -L/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/lib -lc10 -ltorch -ltorch_cpu -ltorch_python -lcudart -lc10_cuda -ltorch_cuda -o build/lib.linux-x86_64-cpython-311/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  copying build/lib.linux-x86_64-cpython-311/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/flash_attn
  creating build/bdist.linux-x86_64/wheel/flash_attn/ops
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/activations.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/rms_norm.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/fused_dense.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/layer_norm.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops
  creating build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/cross_entropy.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/rotary.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/mlp.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/k_activations.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/layer_norm.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  copying build/lib.linux-x86_64-cpython-311/flash_attn/ops/triton/linear.py -> build/bdist.linux-x86_64/wheel/flash_attn/ops/triton
  creating build/bdist.linux-x86_64/wheel/flash_attn/modules
  copying build/lib.linux-x86_64-cpython-311/flash_attn/modules/mha.py -> build/bdist.linux-x86_64/wheel/flash_attn/modules
  copying build/lib.linux-x86_64-cpython-311/flash_attn/modules/mlp.py -> build/bdist.linux-x86_64/wheel/flash_attn/modules
  copying build/lib.linux-x86_64-cpython-311/flash_attn/modules/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/modules
  copying build/lib.linux-x86_64-cpython-311/flash_attn/modules/embedding.py -> build/bdist.linux-x86_64/wheel/flash_attn/modules
  copying build/lib.linux-x86_64-cpython-311/flash_attn/modules/block.py -> build/bdist.linux-x86_64/wheel/flash_attn/modules
  copying build/lib.linux-x86_64-cpython-311/flash_attn/flash_blocksparse_attention.py -> build/bdist.linux-x86_64/wheel/flash_attn
  copying build/lib.linux-x86_64-cpython-311/flash_attn/flash_attn_interface.py -> build/bdist.linux-x86_64/wheel/flash_attn
  copying build/lib.linux-x86_64-cpython-311/flash_attn/flash_attn_triton_og.py -> build/bdist.linux-x86_64/wheel/flash_attn
  copying build/lib.linux-x86_64-cpython-311/flash_attn/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn
  creating build/bdist.linux-x86_64/wheel/flash_attn/utils
  copying build/lib.linux-x86_64-cpython-311/flash_attn/utils/distributed.py -> build/bdist.linux-x86_64/wheel/flash_attn/utils
  copying build/lib.linux-x86_64-cpython-311/flash_attn/utils/benchmark.py -> build/bdist.linux-x86_64/wheel/flash_attn/utils
  copying build/lib.linux-x86_64-cpython-311/flash_attn/utils/pretrained.py -> build/bdist.linux-x86_64/wheel/flash_attn/utils
  copying build/lib.linux-x86_64-cpython-311/flash_attn/utils/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/utils
  copying build/lib.linux-x86_64-cpython-311/flash_attn/utils/generation.py -> build/bdist.linux-x86_64/wheel/flash_attn/utils
  creating build/bdist.linux-x86_64/wheel/flash_attn/layers
  copying build/lib.linux-x86_64-cpython-311/flash_attn/layers/patch_embed.py -> build/bdist.linux-x86_64/wheel/flash_attn/layers
  copying build/lib.linux-x86_64-cpython-311/flash_attn/layers/rotary.py -> build/bdist.linux-x86_64/wheel/flash_attn/layers
  copying build/lib.linux-x86_64-cpython-311/flash_attn/layers/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/layers
  copying build/lib.linux-x86_64-cpython-311/flash_attn/flash_blocksparse_attn_interface.py -> build/bdist.linux-x86_64/wheel/flash_attn
  creating build/bdist.linux-x86_64/wheel/flash_attn/losses
  copying build/lib.linux-x86_64-cpython-311/flash_attn/losses/cross_entropy.py -> build/bdist.linux-x86_64/wheel/flash_attn/losses
  copying build/lib.linux-x86_64-cpython-311/flash_attn/losses/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/losses
  copying build/lib.linux-x86_64-cpython-311/flash_attn/flash_attn_triton.py -> build/bdist.linux-x86_64/wheel/flash_attn
  copying build/lib.linux-x86_64-cpython-311/flash_attn/fused_softmax.py -> build/bdist.linux-x86_64/wheel/flash_attn
  copying build/lib.linux-x86_64-cpython-311/flash_attn/bert_padding.py -> build/bdist.linux-x86_64/wheel/flash_attn
  creating build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/falcon.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/btlm.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/vit.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/__init__.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/bert.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/baichuan.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/bigcode.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/opt.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/gpt_neox.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/gpt.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/llama.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  copying build/lib.linux-x86_64-cpython-311/flash_attn/models/gptj.py -> build/bdist.linux-x86_64/wheel/flash_attn/models
  running install_egg_info
  Copying flash_attn.egg-info to build/bdist.linux-x86_64/wheel/flash_attn-2.5.8-py3.11.egg-info
  running install_scripts
  creating build/bdist.linux-x86_64/wheel/flash_attn-2.5.8.dist-info/WHEEL
  creating '/tmp/pip-wheel-7_xhpf61/.tmp-4qdnj8g3/flash_attn-2.5.8-cp311-cp311-linux_x86_64.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
  adding 'flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so'
  adding 'flash_attn/__init__.py'
  adding 'flash_attn/bert_padding.py'
  adding 'flash_attn/flash_attn_interface.py'
  adding 'flash_attn/flash_attn_triton.py'
  adding 'flash_attn/flash_attn_triton_og.py'
  adding 'flash_attn/flash_blocksparse_attention.py'
  adding 'flash_attn/flash_blocksparse_attn_interface.py'
  adding 'flash_attn/fused_softmax.py'
  adding 'flash_attn/layers/__init__.py'
  adding 'flash_attn/layers/patch_embed.py'
  adding 'flash_attn/layers/rotary.py'
  adding 'flash_attn/losses/__init__.py'
  adding 'flash_attn/losses/cross_entropy.py'
  adding 'flash_attn/models/__init__.py'
  adding 'flash_attn/models/baichuan.py'
  adding 'flash_attn/models/bert.py'
  adding 'flash_attn/models/bigcode.py'
  adding 'flash_attn/models/btlm.py'
  adding 'flash_attn/models/falcon.py'
  adding 'flash_attn/models/gpt.py'
  adding 'flash_attn/models/gpt_neox.py'
  adding 'flash_attn/models/gptj.py'
  adding 'flash_attn/models/llama.py'
  adding 'flash_attn/models/opt.py'
  adding 'flash_attn/models/vit.py'
  adding 'flash_attn/modules/__init__.py'
  adding 'flash_attn/modules/block.py'
  adding 'flash_attn/modules/embedding.py'
  adding 'flash_attn/modules/mha.py'
  adding 'flash_attn/modules/mlp.py'
  adding 'flash_attn/ops/__init__.py'
  adding 'flash_attn/ops/activations.py'
  adding 'flash_attn/ops/fused_dense.py'
  adding 'flash_attn/ops/layer_norm.py'
  adding 'flash_attn/ops/rms_norm.py'
  adding 'flash_attn/ops/triton/__init__.py'
  adding 'flash_attn/ops/triton/cross_entropy.py'
  adding 'flash_attn/ops/triton/k_activations.py'
  adding 'flash_attn/ops/triton/layer_norm.py'
  adding 'flash_attn/ops/triton/linear.py'
  adding 'flash_attn/ops/triton/mlp.py'
  adding 'flash_attn/ops/triton/rotary.py'
  adding 'flash_attn/utils/__init__.py'
  adding 'flash_attn/utils/benchmark.py'
  adding 'flash_attn/utils/distributed.py'
  adding 'flash_attn/utils/generation.py'
  adding 'flash_attn/utils/pretrained.py'
  adding 'flash_attn-2.5.8.dist-info/AUTHORS'
  adding 'flash_attn-2.5.8.dist-info/LICENSE'
  adding 'flash_attn-2.5.8.dist-info/METADATA'
  adding 'flash_attn-2.5.8.dist-info/WHEEL'
  adding 'flash_attn-2.5.8.dist-info/top_level.txt'
  adding 'flash_attn-2.5.8.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  Building wheel for flash_attn (pyproject.toml): finished with status 'done'
  Created wheel for flash_attn: filename=flash_attn-2.5.8-cp311-cp311-linux_x86_64.whl size=161615679 sha256=019cabb84a0f37b55ff08e14959ee34184e658cbc5d6edc3622e44779820b595
  Stored in directory: /tmp/pip-ephem-wheel-cache-g138790y/wheels/d8/91/a6/6160216e602ad906106939541f06f84e2dbc50fbc04b44036d
Successfully built flash_attn
Installing collected packages: flash_attn

Successfully installed flash_attn-2.5.8
Removed build tracker: '/tmp/pip-build-tracker-r7fvrno4'

Resource usage statistics from building flash-attn:
   Process count: 8
   CPU time: Sys=0:03:26.0, User=4:47:19.4
   Memory: 5.6G
   Disk usage: 1.1M
   Time elapsed: 4:56:11.4


Packaging flash-attn
/opt/conda/lib/python3.10/site-packages/conda_build/environ.py:558: UserWarning: The environment variable 'MAX_JOBS' is being passed through with value '1'.  If you are splitting build and test phases with --no-test, please ensure that this value is also set similarly at test time.
  warnings.warn(
/opt/conda/lib/python3.10/site-packages/conda_build/environ.py:558: UserWarning: The environment variable 'TORCH_CUDA_ARCH_LIST' is being passed through with value '8.6+PTX'.  If you are splitting build and test phases with --no-test, please ensure that this value is also set similarly at test time.
  warnings.warn(
Packaging flash-attn-2.5.8-py311h379968c_0
compiling .pyc files...
number of files: 104
Warning: rpath /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/lib is outside prefix /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_ (removing it)
   INFO: sysroot: '/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/zone1970.tab', '/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/zone.tab', '/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/tzdata.zi', '/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/Zulu']'
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libc10.so found in conda-forge/linux-64::libtorch==2.1.2=cuda120_h2aa5df7_303
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libtorch_cpu.so found in conda-forge/linux-64::libtorch==2.1.2=cuda120_h2aa5df7_303
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/python3.11/site-packages/torch/lib/libtorch_python.so found in conda-forge/linux-64::pytorch==2.1.2=cuda120_py311h25b6552_303
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libcudart.so.12 found in conda-forge/linux-64::cuda-cudart==12.0.107=hd3aeb46_8
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libc10_cuda.so found in conda-forge/linux-64::libtorch==2.1.2=cuda120_h2aa5df7_303
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libtorch_cuda.so found in conda-forge/linux-64::libtorch==2.1.2=cuda120_h2aa5df7_303
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx-ng==13.2.0=hc0a3c3a_7
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc-ng==13.2.0=h77fa898_7
   INFO (flash-attn,lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_14
WARNING (flash-attn): interpreter (Python) package conda-forge/linux-64::python==3.11.9=hb806964_0_cpython in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
Fixing permissions
Packaged license file/s.
INFO :: Time taken to mark (prefix)
        0 replacements in 0 files was 1.18 seconds
Files containing CONDA_PREFIX
-----------------------------
lib/python3.11/site-packages/flash_attn_2_cuda.cpython-311-x86_64-linux-gnu.so (binary): Patching
WARNING: Importing conda-verify failed.  Please be sure to test your packages.  conda install conda-verify to make this message go away.
TEST START: /home/conda/staged-recipes/build_artifacts/linux-64/flash-attn-2.5.8-py311h379968c_0.conda
Renaming work directory '/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work' to '/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work_moved_flash-attn-2.5.8-py311h379968c_0_linux-64'
shutil.move(work)=/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work, dest=/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/work_moved_flash-attn-2.5.8-py311h379968c_0_linux-64)
Reloading output folder (local): ...working... done
Solving environment (_test_env): ...working... done

## Package Plan ##

  environment location: /home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place


The following NEW packages will be INSTALLED:

    _libgcc_mutex:        0.1-conda_forge                 conda-forge
    _openmp_mutex:        4.5-2_kmp_llvm                  conda-forge
    bzip2:                1.0.8-hd590300_5                conda-forge
    ca-certificates:      2024.2.2-hbcca054_0             conda-forge
    cuda-cudart:          12.4.127-hd3aeb46_0             conda-forge
    cuda-cudart_linux-64: 12.4.127-h59595ed_0             conda-forge
    cuda-nvrtc:           12.4.127-hd3aeb46_1             conda-forge
    cuda-nvtx:            12.4.127-h59595ed_1             conda-forge
    cuda-version:         12.4-h3060b56_3                 conda-forge
    cudnn:                8.9.7.29-h092f7fd_3             conda-forge
    einops:               0.8.0-pyhd8ed1ab_0              conda-forge
    filelock:             3.14.0-pyhd8ed1ab_0             conda-forge
    flash-attn:           2.5.8-py311h379968c_0           local      
    fsspec:               2024.3.1-pyhca7485f_0           conda-forge
    gmp:                  6.3.0-h59595ed_1                conda-forge
    gmpy2:                2.1.5-py311he48d604_0           conda-forge
    icu:                  73.2-h59595ed_0                 conda-forge
    jinja2:               3.1.3-pyhd8ed1ab_0              conda-forge
    ld_impl_linux-64:     2.40-h55db66e_0                 conda-forge
    libabseil:            20230802.1-cxx17_h59595ed_0     conda-forge
    libblas:              3.9.0-22_linux64_openblas       conda-forge
    libcblas:             3.9.0-22_linux64_openblas       conda-forge
    libcublas:            12.4.5.8-hd3aeb46_1             conda-forge
    libcufft:             11.2.1.3-hd3aeb46_1             conda-forge
    libcurand:            10.3.5.147-hd3aeb46_1           conda-forge
    libcusolver:          11.6.1.9-hd3aeb46_1             conda-forge
    libcusparse:          12.3.1.170-hd3aeb46_1           conda-forge
    libexpat:             2.6.2-h59595ed_0                conda-forge
    libffi:               3.4.2-h7f98852_5                conda-forge
    libgcc-ng:            13.2.0-h77fa898_7               conda-forge
    libgfortran-ng:       13.2.0-h69a702a_7               conda-forge
    libgfortran5:         13.2.0-hca663fb_7               conda-forge
    libhwloc:             2.10.0-default_h2fb2949_1000    conda-forge
    libiconv:             1.17-hd590300_2                 conda-forge
    liblapack:            3.9.0-22_linux64_openblas       conda-forge
    libmagma:             2.7.2-h173bb3b_2                conda-forge
    libmagma_sparse:      2.7.2-h173bb3b_3                conda-forge
    libnsl:               2.0.1-hd590300_0                conda-forge
    libnvjitlink:         12.4.127-hd3aeb46_1             conda-forge
    libopenblas:          0.3.27-pthreads_h413a1c8_0      conda-forge
    libprotobuf:          4.25.1-hf27288f_2               conda-forge
    libsqlite:            3.45.3-h2797004_0               conda-forge
    libstdcxx-ng:         13.2.0-hc0a3c3a_7               conda-forge
    libtorch:             2.1.2-cuda120_h2aa5df7_303      conda-forge
    libuuid:              2.38.1-h0b41bf4_0               conda-forge
    libuv:                1.48.0-hd590300_0               conda-forge
    libxcrypt:            4.4.36-hd590300_1               conda-forge
    libxml2:              2.12.6-h232c23b_2               conda-forge
    libzlib:              1.2.13-hd590300_5               conda-forge
    llvm-openmp:          18.1.5-ha31de31_0               conda-forge
    magma:                2.7.2-h51420fd_3                conda-forge
    markupsafe:           2.1.5-py311h459d7ec_0           conda-forge
    mkl:                  2023.2.0-h84fe81f_50496         conda-forge
    mpc:                  1.3.1-hfe3b2da_0                conda-forge
    mpfr:                 4.2.1-h9458935_1                conda-forge
    mpmath:               1.3.0-pyhd8ed1ab_0              conda-forge
    nccl:                 2.21.5.1-h3a97aeb_0             conda-forge
    ncurses:              6.4.20240210-h59595ed_0         conda-forge
    networkx:             3.3-pyhd8ed1ab_1                conda-forge
    numpy:                1.26.4-py311h64a7726_0          conda-forge
    openssl:              3.3.0-hd590300_0                conda-forge
    pip:                  24.0-pyhd8ed1ab_0               conda-forge
    python:               3.11.9-hb806964_0_cpython       conda-forge
    python_abi:           3.11-4_cp311                    conda-forge
    pytorch:              2.1.2-cuda120_py311h25b6552_303 conda-forge
    readline:             8.2-h8228510_1                  conda-forge
    setuptools:           69.5.1-pyhd8ed1ab_0             conda-forge
    sleef:                3.5.1-h9b69904_2                conda-forge
    sympy:                1.12-pypyh9d50eac_103           conda-forge
    tbb:                  2021.12.0-h00ab1b0_0            conda-forge
    tk:                   8.6.13-noxft_h4845f30_101       conda-forge
    typing_extensions:    4.11.0-pyha770c72_0             conda-forge
    tzdata:               2024a-h0c530f3_0                conda-forge
    wheel:                0.43.0-pyhd8ed1ab_1             conda-forge
    xz:                   5.2.6-h166bdaf_0                conda-forge
    zstd:                 1.5.6-ha6fb4c9_0                conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... By downloading and using the cuDNN conda packages, you accept the terms and conditions of the NVIDIA cuDNN EULA -
  https://docs.nvidia.com/deeplearning/cudnn/sla/index.html

done
export PREFIX=/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place
export SRC_DIR=/home/conda/staged-recipes/build_artifacts/flash-attn_1715047963337/test_tmp
import: 'flash_attn'
import: 'flash_attn'
+ pip check
No broken requirements found.
+ exit 0

Resource usage statistics from testing flash-attn:
   Process count: 4
   CPU time: Sys=0:00:00.3, User=0:00:01.1
   Memory: 337.3M
   Disk usage: 24B
   Time elapsed: 0:00:06.0


TEST END: /home/conda/staged-recipes/build_artifacts/linux-64/flash-attn-2.5.8-py311h379968c_0.conda
```
</details>

@carterbox, did you want to keep using the simplified `setup.py` and `pyproject.toml` file, or try reverting back to upstream one?

Reply:
>  but it's because it continued to try to build for another Python version (Python 3.10) for some reason (maybe because we're not using noarch).

In staged recipes, there is only one runner per platform, every python variant is built on the same runner.

> @carterbox, did you want to keep using the simplified setup.py and pyproject.toml file, or try reverting back to upstream one?

I want to keep the simplified scripts for now. The only drawback is we have to manually update the source files list, some compile args, and dependencies. However, I think that is better than working around all of special operations upstream has added to their build script which are not compatible with our build environment.

Reply:
@weiji14, once the feedstock is allocated, please do some experiments to see how much time it takes to run with MAX_JOBS=$CPU_COUNT and how many CUDA archs can be added to the arch list. We want to build for as many of 8.0;8.6;8.9;9.0+PTX as we can.

Reply:
Awesome, thanks so much @carterbox! The initial feedstock commit actually failed due to running out of disk space :sweat_smile: But I've opened a PR now at https://github.com/conda-forge/flash-attn-feedstock/pull/1, so we can continue discussion there.